packages/react-menu/src/components/MenuList/renderMenuList.tsx

```diff
@@ -1,12 +1,18 @@
 import * as React from 'react';
 import { getSlots } from '@fluentui/react-utils';
 import { MenuListState } from './MenuList.types';
+import { MenuListProvider } from '../../menuListContext';
 
 /**
  * Function that renders the final JSX of the component
  */
 export const renderMenuList = (state: MenuListState) => {
   const { slots, slotProps } = getSlots(state);
+  const { onCheckedValueChange, checkedValues } = state;
 
-  return <slots.root {...slotProps.root}>{state.children}</slots.root>;
+  return (
+    <MenuListProvider value={{ onCheckedValueChange, checkedValues }}>
```

Comment:
React should use `react-context-selector` there as each render of this context will produce React reconcilation:
- `value` is unstable, we can solve it with `useMemo()`, but... =>
- `checked` values will still change frequently

Reply:
Can I add a TODO and come back to it, it seems complex enough that I should revisit and base a PR around that

Reply:
Yes, TODO with an explanation works 👍

Reply:
I've added the TODO in `menuListContext`